plugin/pkg/scheduler/schedulercache/cache.go

```diff
@@ -95,7 +95,7 @@ func (cache *schedulerCache) UpdateNodeNameToInfoMap(nodeNameToInfo map[string]*
 func (cache *schedulerCache) List(selector labels.Selector) ([]*v1.Pod, error) {
 	cache.mu.Lock()
 	defer cache.mu.Unlock()
-	var pods []*v1.Pod
+	pods := make([]*v1.Pod, 0, len(cache.podStates))
```

Comment:
This allocates memory enough to hold all the pods, but in practice only a fractions of pods are probably matched against the selector and added to the array. So, the new code probably wastes memory, which may or may not be justifiable given the performance improvement. That brings up back to the point that other reviewers mentioned as well. How much performance improvement do we get out of this change?